### PR TITLE
pkg/fileutil: optimize file stats error

### PR DESCRIPTION
**Pull request by tangcong**
when I try to upgrade etcd cluster from v3.4 to v3.5, I failed to start etcd etcd process.  I have to find solution from code, pr #11798 introduces this issue.  it is better if we can print detail desired file permission in error log and add breaking change doc for it. thanks. @spzala 
```
Jun 10 13:50:45 VM-test-ubuntu bash[11974]: {"level":"fatal","ts":"2020-06-10T13:50:45.510+0800","caller":"etcdmain/etcd.go:197","msg":"discovery failed","error":"cannot access data directory: directory \"/data/etcd.data\" exist without desired file permission. \"drwxr-xr-x\"","
```

### CODE REVIEW

File: CHANGELOG-3.5.md [hunk 1]
```diff
@@ -60,6 +60,9 @@ See [code changes](https://github.com/etcd-io/etcd/compare/v3.4.0...v3.5.0) and
 - Changed `pkg/flags` function signature to [support structured logger](https://github.com/etcd-io/etcd/pull/11616).
   - Previously, `SetFlagsFromEnv(prefix string, fs *flag.FlagSet) error`, now `SetFlagsFromEnv(lg *zap.Logger, prefix string, fs *flag.FlagSet) error`.
   - Previously, `SetPflagsFromEnv(prefix string, fs *pflag.FlagSet) error`, now `SetPflagsFromEnv(lg *zap.Logger, prefix string, fs *pflag.FlagSet) error`.
+- Changed behavior of [data dir permission](https://github.com/etcd-io/etcd/pull/11798).
```
[the remark this answers is not in the document]
Comment (tangcong) [reply CHANGELOG-3.5.md:63]: updated. thanks.

**Comment by spzala** [comment]
CI failure is not related.

**Comment by spzala** [comment]
@tangcong can you please cherry pick this to 3.4 and 3.3? Thanks!

**Comment by tangcong** [comment]
> @tangcong can you please cherry pick this to 3.4 and 3.3? Thanks!

it is minor changes. can you merge it in pr #12081 and #12082 ? thanks.

**Comment by tangcong** [comment]
@spzala pr #12089 #12090. done

**Comment by spzala** [comment]
> @spzala pr #12089 #12090. done

Thanks @tangcong

**Pull request merged**